app/lib/actions/katello/sync_plan/update.rb

```diff
@@ -4,16 +4,43 @@ module SyncPlan
       class Update < Actions::EntryAction
         def plan(sync_plan, sync_plan_params = nil)
           action_subject(sync_plan)
+          rec_logic_change = sync_plan.sync_date != sync_plan_params["sync_date"] || sync_plan.interval != sync_plan_params["interval"] ||
+              sync_plan.enabled != sync_plan_params["enabled"]
+          sync_date = sync_plan_params.try(:[], "sync_date").try(:to_time)
+          fail _("Sync Date is probably null" + sync_date.to_s) if sync_date.nil?
           sync_plan.update_attributes(sync_plan_params) if sync_plan_params
```

Comment:
You could simplify some of the above logic with:
```
sync_plan.update_attributes(sync_plan_params) 
rec_logic_change = sync_plan.saved_change_to_attribute(:sync_date) || sync_plan.saved_change_to_attribute(:interval)
```


I would also use `.update_attributes!()` so that an exception gets thrown if there is an error

Reply:
You might also consider if we can just push this to the model as after_save callbacks, and then do we even need an update action?